### PR TITLE
Admin view

### PR DESCRIPTION
**Pull request by BojithaPiyathilake**

### CODE REVIEW

File: frontend/src/components/MainComponent.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+						//editUser = {this.props.editUser}
+					/>
+				</div>
+			</div>
+	  )
+		
+	}
+	  
+	}
+export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Main));
```
Comment (yashodgayashan) [inline frontend/src/components/MainComponent.js:61]:
```suggestion
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Main));	

```

File: frontend/src/components/MainComponent.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+					<AddModal postUser={this.props.postUser}/>
+				</div>
+				
+				<div className="col-12 mt-2">
+					<TableComponent 
+						users={this.props.users.users}
+						isLoading={this.props.users.isLoading}
+						errMess={this.props.users.errMess}
+						deleteUser = {this.props.deleteUser}
+						//editUser = {this.props.editUser}
```
Comment (yashodgayashan) [inline frontend/src/components/MainComponent.js:52]: We are nor committing commented codes as a good practice.

File: frontend/src/redux/ActionCreators.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
+import * as ActionTypes from './ActionTypes';
+import { baseUrl } from '../shared/baseUrl';
+
+export const addUser = (user) => ({	
+    type: ActionTypes.ADD_USER,
+    payload: user
+});
+
+
+export const fetchUsers = () => (dispatch) => {
```
Comment (yashodgayashan) [inline frontend/src/redux/ActionCreators.js:10]: Please explain why there are two arrow functions?

File: frontend/src/redux/ActionCreators.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+    dispatch(usersLoading(true));			//sets loading to true
```
Comment (yashodgayashan) [inline frontend/src/redux/ActionCreators.js:12]: Please format the code.

File: frontend/src/redux/configureStore.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+			users: Users,
+			...createForms({		//adds the necessary reducer functions and states for forms
+				feedback: InitialFeedback
+			})
+		}),
+		applyMiddleware (thunk, logger)	
+    );
+
+    return store;
+}
```
Comment (yashodgayashan) [inline frontend/src/redux/configureStore.js:20]:
```suggestion
}

```

File: frontend/src/redux/users.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+		  return {...state, errMess: action.payload};
+
+		case ActionTypes.ADD_USER:
+			var user = action.payload;
+			return { ...state, users: state.users.concat(user)};
+
+		default:
+		  return state;
+	  }
+};
```
Comment (yashodgayashan) [inline frontend/src/redux/users.js:22]:
```suggestion
};

```

File: frontend/src/shared/baseUrl.js [hunk 7]
```diff
@@ -0,0 +1 @@
+export const baseUrl = 'http://localhost:3001/';
```
Comment (yashodgayashan) [inline frontend/src/shared/baseUrl.js:1]: Add this to a file naming constants.

**Comment by BojithaPiyathilake** [comment]
Won't be using this branch will be using the one with context API

**Pull request closed without merging**